src/bin/ion/commands/generate/generator.rs

```diff
@@ -501,6 +540,105 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
+        &mut self,
+        constraints: &[IslConstraint],
+        code_gen_context: &mut CodeGenContext,
+        parent_isl_type: &IslType,
+    ) -> CodeGenResult<AbstractDataType> {
+        let mut wrapped_scalar_builder = WrappedScalarBuilder::default();
+        wrapped_scalar_builder
+            .name(self.current_type_fully_qualified_name.to_owned())
+            .source(parent_isl_type.to_owned());
+        for constraint in constraints {
+            match constraint.constraint() {
+                IslConstraintValue::Type(isl_type) => {
+                    let type_name = self
+                        .fully_qualified_type_ref_name(isl_type, code_gen_context)?
+                        .ok_or(invalid_abstract_data_type_raw_error(format!(
+                            "Could not determine `FullQualifiedTypeReference` for type {:?}",
+                            isl_type
+                        )))?;
+
+                    // by default fields aren't closed
+                    wrapped_scalar_builder.base_type(type_name);
+                }
+                _ => {
+                    return invalid_abstract_data_type_error(
+                        "Could not determine the abstract data type due to conflicting constraints",
+                    );
+                }
+            }
+        }
+
+        Ok(AbstractDataType::WrappedScalar(
+            wrapped_scalar_builder.build()?,
+        ))
+    }
+
+    /// Builds `AbstractDataType::Scalar` from the given constraints.
+    /// ```
+    /// { type: string }
+    /// ```
+    /// This method builds `AbstractDataTYpe`as following:
+    /// ```
+    /// AbstractDataType::Scalar(
+    ///  Scalar {
+    ///     base_type: FullyQualifiedTypeReference { type_name: vec!["String"], parameters: vec![] }
+    ///     doc_comment: None // There is no doc comment defined in above ISL type def
+    ///     source: IslType { .. } // Represents the `IslType` that is getting converted to `AbstractDataType`
+    ///  }
+    /// )
+    /// ```
+    fn build_scalar_from_constraints(
+        &mut self,
+        constraints: &[IslConstraint],
+        code_gen_context: &mut CodeGenContext,
+        parent_isl_type: &IslType,
+    ) -> CodeGenResult<AbstractDataType> {
+        let mut scalar_builder = ScalarBuilder::default();
+        scalar_builder.source(parent_isl_type.to_owned());
+        for constraint in constraints {
+            match constraint.constraint() {
+                IslConstraintValue::Type(isl_type) => {
+                    let type_name = self
+                        .fully_qualified_type_ref_name(isl_type, code_gen_context)?
+                        .ok_or(invalid_abstract_data_type_raw_error(
+                            "Could not determine `FullQualifiedTypeReference` for `struct`, `list` or `sexp` as open ended container types aren't supported."
+                        ))?;
+
+                    // by default fields aren't closed
```

Comment:
What does this comment mean related to scalars values?

Reply:
Good catch! Removed it.